plugins/net_plugin/net_plugin.cpp

```diff
@@ -974,7 +974,9 @@ namespace eosio {
          fetch_wait( );
       }
       else {
-         close_after_send = m.contains<go_away_message>();
+         if (m.contains<go_away_message>()) {
+            close_after_send = m.get<go_away_message>().reason;
+         }
```

Comment:
Purpose of go_away_message in `protocol.hpp` is to specify the error reason and node_id cause the error